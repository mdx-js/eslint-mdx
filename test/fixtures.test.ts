import path from 'path'

import { ESLint } from 'eslint'

const cli = new ESLint({
  ignore: false,
  fix: true,
  useEslintrc: false,
  baseConfig: {
    parserOptions: {
      ecmaVersion: 2020,
      sourceType: 'module',
    },
    extends: ['plugin:mdx/recommended'],
    plugins: ['react', 'prettier'],
  },
})

describe('fixtures', () => {
  it('should match all snapshots', async () => {
    const results = await cli.lintFiles(['test/fixtures/*.{md,mdx}'])
    for (const { filePath, messages } of results) {
      expect(messages).toMatchSnapshot(path.basename(filePath))
    }
  })
})
