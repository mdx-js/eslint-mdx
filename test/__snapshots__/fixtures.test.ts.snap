// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures lint code blocks should work as expected: code-blocks.md 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "line": 3,
    "message": "Do not use headings with similar content (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-duplicate-headings",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "line": 3,
    "message": "Don’t use multiple top level headings (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 36,
    "line": 36,
    "message": "Do not use headings with similar content (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-duplicate-headings",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 36,
    "line": 36,
    "message": "Don’t use multiple top level headings (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 48,
    "line": 48,
    "message": "Do not use headings with similar content (5:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-duplicate-headings",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 48,
    "line": 48,
    "message": "Don’t use multiple top level headings (5:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
]
`;

exports[`fixtures should match all snapshots: 287.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: 292.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: 334.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: adjacent.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: basic.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: blank-lines.mdx 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 16,
    "endLine": 27,
    "line": 27,
    "message": "Don’t use multiple top level headings (3:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
  Object {
    "column": 2,
    "endColumn": 7,
    "endLine": 31,
    "line": 31,
    "message": "'Tweet' is not defined.",
    "messageId": "undefined",
    "nodeType": "JSXIdentifier",
    "ruleId": "react/jsx-no-undef",
    "severity": 2,
  },
  Object {
    "column": 2,
    "endColumn": 14,
    "endLine": 33,
    "line": 33,
    "message": "'TextGradient' is not defined.",
    "messageId": "undefined",
    "nodeType": "JSXIdentifier",
    "ruleId": "react/jsx-no-undef",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 36,
    "endLine": 35,
    "line": 35,
    "message": "Don’t use multiple top level headings (3:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
  Object {
    "column": 7,
    "line": 35,
    "message": "\`'\` can be escaped with \`&apos;\`, \`&lsquo;\`, \`&#39;\`, \`&rsquo;\`.",
    "messageId": "unescapedEntityAlts",
    "nodeType": "JSXText",
    "ruleId": "react/no-unescaped-entities",
    "severity": 1,
  },
  Object {
    "column": 2,
    "endColumn": 9,
    "endLine": 41,
    "line": 41,
    "message": "'YouTube' is not defined.",
    "messageId": "undefined",
    "nodeType": "JSXIdentifier",
    "ruleId": "react/jsx-no-undef",
    "severity": 2,
  },
]
`;

exports[`fixtures should match all snapshots: code-blocks.md 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "line": 3,
    "message": "Do not use headings with similar content (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-duplicate-headings",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "line": 3,
    "message": "Don’t use multiple top level headings (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
]
`;

exports[`fixtures should match all snapshots: details.mdx 1`] = `
Array [
  Object {
    "column": 290,
    "line": 3,
    "message": "\`'\` can be escaped with \`&apos;\`, \`&lsquo;\`, \`&#39;\`, \`&rsquo;\`.",
    "messageId": "unescapedEntityAlts",
    "nodeType": "JSXText",
    "ruleId": "react/no-unescaped-entities",
    "severity": 1,
  },
  Object {
    "column": 5,
    "line": 5,
    "message": "\`'\` can be escaped with \`&apos;\`, \`&lsquo;\`, \`&#39;\`, \`&rsquo;\`.",
    "messageId": "unescapedEntityAlts",
    "nodeType": "JSXText",
    "ruleId": "react/no-unescaped-entities",
    "severity": 1,
  },
]
`;

exports[`fixtures should match all snapshots: jsx-in-list.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: leading-spaces.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: markdown.md 1`] = `Array []`;

exports[`fixtures should match all snapshots: no-jsx-html-comments.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: no-unescaped-entities.mdx 1`] = `
Array [
  Object {
    "column": 8,
    "line": 2,
    "message": "\`>\` can be escaped with \`&gt;\`.",
    "messageId": "unescapedEntityAlts",
    "nodeType": "JSXText",
    "ruleId": "react/no-unescaped-entities",
    "severity": 1,
  },
  Object {
    "column": 13,
    "line": 5,
    "message": "\`>\` can be escaped with \`&gt;\`.",
    "messageId": "unescapedEntityAlts",
    "nodeType": "JSXText",
    "ruleId": "react/no-unescaped-entities",
    "severity": 1,
  },
]
`;

exports[`fixtures should match all snapshots: processor.mdx 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "line": 3,
    "message": "Do not use headings with similar content (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-duplicate-headings",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "line": 3,
    "message": "Don’t use multiple top level headings (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
]
`;

exports[`fixtures should match all snapshots: remark.md 1`] = `Array []`;

exports[`fixtures should match all snapshots: remark.mdx 1`] = `Array []`;
